cunumeric/module.py

```diff
@@ -2092,17 +2167,15 @@ def repeat(a, repeats, axis=None):
         if repeats == 0:
             empty_shape = list(array.shape)
             empty_shape[axis] = 0
-            empty_shape = tuple(empty_shape)
-            return ndarray(shape=empty_shape, dtype=array.dtype)
+            return ndarray(shape=tuple(empty_shape), dtype=array.dtype)
         # repeats should be integer type
         if not isinstance(repeats, int):
             runtime.warn(
                 "converting repeats to an integer type",
                 category=UserWarning,
             )
-        repeats = np.int64(repeats)
         result = array._thunk.repeat(
-            repeats=repeats,
+            repeats=np.int64(repeats),
```

Comment:
Moving to the call site side-steps having to deal with a type-widening re-assigment.